Cognigy/WebchatPlugins
# Feature/19538 add upload to live agent through webchat

This PR adds support for file uploading to the Live Agent.

How to test:

1. Create a conversation with handover provider Live Agent and get sure you can exchange messages between user and agent
2. Start your chrome browser in unsafe mode to avoid CORS issues with 'google-chrome --disable-web-security --user-data-dir="~"'
3. Configure your webchat widget to use a fixed userId and sessionId and chat with the agent. Check requests in the network tab to retrieve your URLToken
4. Build and import the file-upload plugin from this package after the webchat.js or webchat.legacy.js file in the file you use to call the widget.
5. Either fork or export and import the Chatwoot APIs from postman https://www.postman.com/chatwoot/workspace/chatwoot-apis/overview
6. Create/update the following environment variables:
- host = https://dev-live-agent.cognigy.ai
- api_version = api/v1
- api_access_token = Access Token from your Live Agent Profile Settings
- account_id = Your Live Agent Account Id, can be taken from the URL
7. Call "Collections-->Chatwoot Application API v1.0-->Conversations-->List all conversations".
8. Search the JSON output from the previous step. Add a new environment variable called contact_id and assign it the value from  data.payload[0].meta.sender.id
9. Call "Collections-->Chatwoot Application API v1.0-->Contacts-->Get Contact" and note the value from payload.contact_inboxes[0].source_id as client_identifier.
10. Use the notify API from service-api and send such a request to your conversation:

{
  "userId": "1", //<--same as in your webchat widget config
  "text": "",
  "data": {
"_plugin": {
"type": "file-upload",
"service": "live-agent",
"inboxIdentifier":"RC1r4dqhQxiVHQrpQjHZKoxm",//<-- taken from the configuration of your Live Agent inbox(not the number in the URL)
"accountId":"7", //<--your user account in Live Agent, can be taken from the URL
"conversationId":"28", //<--your conversationId in Live Agent, can be taken from the URL
"contactIdentifier":"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", //<--retrieved in step 9
"host":"https://dev-live-agent.cognigy.ai"}}, //<--the hostname of your Live Agent installation
  "URLToken": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx", //<--taken from network tab while in conversation
  "sessionId": "1" //<--same as in your webchat widget config
}

11. You should now receive a file upload button that you can press for uploading. Select a picture and it should get uploaded.
12. The picture should be available in the Live Agent afterwards
13. Upload files with these extensions: .png, jpg, jpeg, pdf, all should be rendered as images, except for the pdf, it should have a download button.

Our Live Agent installation is facing some issues when downloading files using the url we received, therefore I discussed with Nacho that this is out of scope and will be solved by his team
